src/cubi_tk/sodar_api.py

```diff
@@ -1,47 +1,32 @@
 import argparse
-from collections import namedtuple
 from functools import reduce
-import os
 from typing import Literal
 import urllib.parse as urlparse
 
 from loguru import logger
 import requests
 
-from .common import is_uuid, load_toml_config
+from cubi_tk.parsers import check_args_sodar_config_parser
+
+from .common import is_uuid
 from .exceptions import ParameterException, SodarAPIException
 
 
 class SodarAPI:
-    def __init__(self, sodar_url: str, sodar_api_token: str, project_uuid: str):
-        self.sodar_url = sodar_url
+    def __init__(self, sodar_server_url: str, sodar_api_token: str, project_uuid: str, config = None):
+        self.sodar_server_url = sodar_server_url
```

Comment:
maybe there's an argument to made here, to just pass the "args" object (which now should always use the same structure for sodar related arguments)